src/data_provider/src/extended_sources/users/src/users_windows_utils_wrapper.cpp

```diff
@@ -0,0 +1,520 @@
+/* Copyright (C) 2015, Wazuh Inc.
+ * All rights reserved.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#include "users_windows_utils_wrapper.hpp"
+#include "windows_api_wrapper.hpp"
+
+UsersHelper::UsersHelper(
+    std::shared_ptr<IWindowsApiWrapper> winapiWrapper)
+    : m_winapiWrapper(std::move(winapiWrapper)) {}
+
+UsersHelper::UsersHelper()
+    : m_winapiWrapper(std::make_shared<WindowsApiWrapper>()) {}
+
+std::string UsersHelper::psidToString(PSID sid)
+{
+    LPSTR sid_out = nullptr;
+    auto ret = m_winapiWrapper->ConvertSidToStringSidAWrapper(sid, &sid_out);
+
+    if (ret == 0)
+    {
+        // std::cout << "ConvertSidToString failed with " << GetLastError();
+        return {};
+    }
+
+    std::string sid_string(sid_out);
+    m_winapiWrapper->FreeSidWrapper(sid_out);
+    return sid_string;
+}
+
+std::string UsersHelper::wstringToString(const wchar_t* src)
```

Comment:
A similat method is available in utils, maybe we can reuse it here:

**_Utils::EncodingWindowsHelper::wstringToStringUTF8_**

Reply:
Done!